storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/RemotePartitionMetadataStore.java

```diff
@@ -91,6 +109,23 @@ public void handleRemotePartitionDeleteMetadata(RemotePartitionDeleteMetadata re
         }
     }
 
+    @Override
+    public void syncLogMetadataDataFile(TopicIdPartition topicIdPartition,
+                                        int metadataPartition,
+                                        Long metadataPartitionOffset) throws IOException {
+        //todo-tier write partitions
+        RemotePartitionDeleteMetadata partitionDeleteMetadata = idToPartitionDeleteMetadata.get(topicIdPartition);
+        if (partitionDeleteMetadata != null) {
+            log.info("Skipping syncing of metadata snapshot as remote partition [{}] is with state: [{}] ", topicIdPartition,
+                     partitionDeleteMetadata);
+        } else {
```

Comment:
RemotePartitionRemover may have the functionality of removing in a different way and revisit that when we have the RemotePartitionRemover functionality.
For now, the cache is stored as part of the topic partition log directory and it will be deleted when the respective partition is deleted locally by a broker.